CMakeLists.txt

```diff
@@ -1,18 +1,31 @@
+# Copyright (c) 2020 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
 cmake_minimum_required(VERSION 3.13)
 
-# Mbed OS system root path - that probably needs to come from tools
-# TODO: @mbed-os-tools
+# TODO: @mbed-os-tools MBED_ROOT and MBED_CONFIG_PATH should probably come from mbedtools
 set(MBED_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/mbed-os CACHE INTERNAL "")
+set(MBED_CONFIG_PATH ${CMAKE_CURRENT_SOURCE_DIR}/.mbedbuild CACHE INTERNAL "")
+set(APP_TARGET mbed-os-example-blinky)
 
-# Include standard Mbed definitions
-include(${MBED_ROOT}/cmake/app.cmake)
+add_subdirectory(${MBED_ROOT})
 
-project(mbed-os-example-blinky)
+add_executable(${APP_TARGET})
 
-# uncomment below to have a verbose build process
-#set(CMAKE_VERBOSE_MAKEFILE ON)
+mbed_os_target_linker_script(${APP_TARGET})
 
-# Sources for main application
-target_sources(app PRIVATE main.cpp)
+project(${APP_TARGET})
+
+target_sources(${APP_TARGET}
+    PRIVATE
+        main.cpp
+)
+
+target_link_libraries(${APP_TARGET} mbed-os)
+
+mbed_os_bin_hex(${APP_TARGET})
```

Comment:
is this actually the post hook? We should generate by default the bin, hex. That was one of the things people wanted as well, there was a PR in the Mbed OS in the recent weeks to do it as well.. If that is the case. therefore I am wondering if this is needed as it should be default behavior. 

Is this needed here ?

Reply:
> is this actually the post hook?

Yes

> We should generate by default the bin, hex

In order to call the pre and post hooks (which use `add_custom_command()`) it needs to be from where the target (the executable`$APP_TARGET`) gets created (therefore the app). Calling the hook within Mbed OS is not possible because the target is not created in Mbed OS)

> Is this needed here ?

Yes

Reply:
ok, I believe this will need to be renamed once tools have pre/post hooks implemented. There will be default one provided by Mbed OS and target can add more (coming from tools) so would be most probably then in the mbedbuild, another file (target_config or similar).

Worth creating a ticket for this alignment